spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/apache/RibbonLoadBalancingHttpClient.java

```diff
@@ -75,12 +75,13 @@ protected HttpClient createDelegate(IClientConfig config) {
 	public RibbonApacheHttpResponse execute(RibbonApacheHttpRequest request,
 			final IClientConfig configOverride) throws Exception {
 		final RequestConfig.Builder builder = RequestConfig.custom();
-		if (configOverride != null) {
-			builder.setConnectTimeout(configOverride.get(
+		IClientConfig config = configOverride != null ? configOverride : this.config;
+		if (config != null) {
```

Comment:
Wondering if config can ever be `null` now?  It doesn't appear to be the case now, I think we can get rid of this `if/else`